Assignment7 Sheqi.out

```diff
@@ -0,0 +1,42 @@
+Assignment 7  Sheqi Zhang
+
+1. Explain the difference between ++*p, *p++ and *++p, if there is any. 
+Answer:
+++*p increments the value at location p points to, then evaluates to incremented value.
+*p++ evaluates the value at location p points to, then advances p.
+*++p increments the value of p, then evaluate the location ++p points to.
+
+2. Is the left to right or right to left order guaranteed for operator precedence?
+Answer:
+In most situations, the left to right order is guarenteed, while for operators like "+=","-=", the order is right to left.
```

Comment:
The simple answer to this question is neither. C doesn’t always evaluate left-to-right or right-to-left. Generally, function calls are evaluated first, followed by complex expressions and then simple expressions.